cloud/amazon/ec2_vpc_route_table.py

```diff
@@ -282,6 +286,18 @@ def route_spec_matches_route(route_spec, route):
         'interface_id': 'interface_id',
         'vpc_peering_connection_id': 'vpc_peering_connection_id',
     }
+
+    # This is a workaround to catch managed NAT gateways as they do not show
+    # up in any of the returned values when describing route tables.
+    # The caveat of doing it this way is that if there was an existing
+    # route for another nat gateway in this route table there is not a way to
+    # change to another nat gateway id. Long term solution would be to utilise
+    # boto3 which is a very big task for this module or to update boto.
+    if route_spec.get('gateway_id') and 'nat-' in route_spec['gateway_id']:
+        if route.destination_cidr_block == route_spec['destination_cidr_block']:
+            if all((not route.gateway_id, not route.instance_id, not route.interface_id, not route.vpc_peering_connection_id)):
```

Comment:
@Etherdaemon I wasn't suggesting that the logic change. `all(not a, not b, not c)` is logically identical to `not any(a, b, c)`. It's just longer and more difficult to read, especially with these long member names. Am I misunderstanding what you're referring to? In any case, it's not a blocking comment.